### PR TITLE
[ProtocolTests] Refactor XML serialization codegen template, 

### PR DESCRIPTION
**Pull request by SergeyRyabinin**
Also fixes fixes wrong timestamp format in a corner case

*Issue #, if available:*
Protocol tests
*Description of changes:*
Refactor how XML serialization logic is generated by creating macro functions that can generate a corresponding code, instead of repeating the logic in the every case.
*Check all that applies:*
- [ ] Did a review by yourself.
- [ ] Added proper tests to cover this PR. (If tests are not applicable, explain.)
- [ ] Checked if this PR is a breaking (APIs have been changed) change.
- [ ] Checked if this PR will _not_ introduce cross-platform inconsistent behavior.
- [ ] Checked if this PR would require a ReadMe/Wiki update.

Check which platforms you have built SDK on to verify the correctness of this PR.
- [ ] Linux
- [ ] Windows
- [ ] Android
- [ ] MacOS
- [ ] IOS
- [ ] Other Platforms

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: generated/protocol-tests/test-clients/aws-cpp-sdk-rest-xml-protocol/source/model/XmlTimestampsRequest.cpp [hunk 1]
```diff
@@ -53,25 +53,25 @@ Aws::String XmlTimestampsRequest::SerializePayload() const
   if(m_epochSecondsHasBeenSet)
   {
    XmlNode epochSecondsNode = parentNode.CreateChildElement("epochSeconds");
-   epochSecondsNode.SetText(m_epochSeconds.ToGmtString(Aws::Utils::DateFormat::ISO_8601));
+   epochSecondsNode.SetText(m_epochSeconds.ToGmtString(Aws::Utils::DateFormat::${timestamptFormatStr}));
```
Comment (sbiscigl) [inline generated/protocol-tests/test-clients/aws-cpp-sdk-rest-xml-protocol/source/model/XmlTimestampsRequest.cpp:56]: `${timestamptFormatStr}` this might be a missing code gen case

File: generated/protocol-tests/tests/input/rest-xml-1/CMakeLists.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 add_project(rest-xml-1-input-protocol-tests
     "Tests for the protocol rest-xml of AWS C++ SDK"
     testing-resources
-    aws-cpp-sdk-rest-xml-protocol-namespace
```
Comment (sbiscigl) [inline generated/protocol-tests/tests/input/rest-xml-1/CMakeLists.txt:4]: should cmake be changing in this revision? some seem to be getting `-namespace` others removing `-namespace` anything up there?

Comment (SergeyRyabinin) [reply generated/protocol-tests/tests/input/rest-xml-1/CMakeLists.txt:4]: It is the issue of the code gen not being stable. Will fix soon.

File: generated/src/aws-cpp-sdk-s3control/source/model/S3ObjectMetadata.cpp [hunk 3]
```diff
@@ -160,6 +160,15 @@ void S3ObjectMetadata::AddToNode(XmlNode& parentNode) const
 
   if(m_userMetadataHasBeenSet)
   {
+   XmlNode userMetadataParentNode = parentNode.CreateChildElement("UserMetadata");
+   for(const auto& mapItem : m_userMetadata)
+   {
+     XmlNode userMetadataMapEntryNode = userMetadataParentNode.CreateChildElement("entry");
+     XmlNode userMetadataKeyNode = userMetadataMapEntryNode.CreateChildElement("key");
```
[the remark this answers is not in the document]
Comment (sbera87) [reply generated/src/aws-cpp-sdk-s3control/source/model/S3ObjectMetadata.cpp:167]: One way I have dealt with this in the past is from java (using recursive function and a context) to generate the code and then just use the code in the template. This will greatly simplify the velocity template and push the computation on the Java side

**Review by sbiscigl — APPROVED** [review]
